### ISSUE TITLE
useFrameSVGAssemblingAnimation hook has been removed from package

### ISSUE THREAD

**Issue by Webitect0x** [romelperez/arwes#225]
Following the documentation can be misleading for individuals who would like to use the useFrameSVGAssemblingAnimation as I believe using the provided animations is a core part of using the arwes library. Seems as if it's been removed even when using older versions.

**Comment by romelperez** [comment]
Hey @Webitect0x ,

Sorry for the delay. It was removed the `@next` version, but it is available in the latest `@alpha` release.

Keep in mind: `@next` version is not official and is for private development. `@alpha` and in the future `@beta` and normal versions are official public release.

Thank you for commenting here.

**Comment by Webitect0x** [comment]
No problem! I've tried installing npm install @arwes/react@1.0.0-alpha.23 and I still get the same error
```
The requested module '/node_modules/.vite/deps/@arwes_react-frames.js?v=b88c370b' does not provide an export named 'useFrameSVGAssemblingAnimation
```
When I check the index.d.ts from @arwes/react-frames it doesn't provide the export anymore I only see these
```
export * from './FrameSVG/index.js';
export * from './FrameSVGOctagon/index.js';
export * from './FrameSVGUnderline/index.js';
export * from './FrameSVGCorners/index.js';
export * from './FrameSVGLines/index.js';
export * from './FrameSVGNefrex/index.js';
export * from './FrameSVGKranox/index.js';
export * from './Illuminator/index.js';
export * from './IlluminatorSVG/index.js';
```

**Comment by romelperez** [comment]
Most likely you are not fixing the version of the packages. It's there in the alpha version: https://www.npmjs.com/package/@arwes/react-frames/v/1.0.0-alpha.23?activeTab=code

Try removing `package-lock.json` or the lock file of the package manager you are using, fix the version in the `package.json`, and then try installing again.

**Comment by Webitect0x** [comment]
When I install the @arwes/react package with 'npm install @arwes/react@1.0.0-alpha.23' It's not there, however if I install @arwes/react-frames with 'npm i @arwes/react-frames@1.0.0-alpha.23' it is included in that package but then I get this error
```
✘ [ERROR] No matching export in "node_modules/@arwes/frames/build/esm/index.js" for import "renderFrameSVGPaths"

    node_modules/@arwes/react-frames/build/esm/FrameSVG/FrameSVG.js:4:9:
      4 │ import { renderFrameSVGPaths } from '@arwes/frames';
```
this is the import code I have 
```
import { useFrameSVGAssemblingAnimation } from "@arwes/react-frames";
```

**Comment by romelperez** [comment]
It seems some packages are in one version and others in another.

Try to remove all ARWES dependencies and just leave just `@arwes/react` like:

```json
{
  "dependencies": {
    "@arwes/react": "1.0.0-alpha.23"
  }
}
```

**Comment by Webitect0x** [comment]
I initialized a new react project, saved the exact version
```
npm install @arwes/react@1.0.0-alpha.23 --save-exact
```
and still
```
Module '"@arwes/react-frames"' has no exported member 'useFrameSVGAssemblingAnimation'.ts(2305)

```

**Comment by 1NF053C** [comment]
@romelperez Could you provide any pointers on how someone could recreate this functionality?

**Comment by romelperez** [comment]
Hey @Webitect0x , @1NF053C ,

Since I wasn't using that version, I didn't run into the problem before and didn't dive into it properly. Because of the internal dependency versions, it was not working even if you installed the main packages with the exact version.

The solution for now is to fix the version of all the ARWES dependencies like this:

```bash
npm i --save-exact \
    arwes@1.0.0-alpha.23 \
    @arwes/animated@1.0.0-alpha.23 \
    @arwes/animator@1.0.0-alpha.23 \
    @arwes/bgs@1.0.0-alpha.23 \
    @arwes/bleeps@1.0.0-alpha.23 \
    @arwes/core@1.0.0-alpha.23 \
    @arwes/frames@1.0.0-alpha.23 \
    @arwes/text@1.0.0-alpha.23 \
    @arwes/theme@1.0.0-alpha.23 \
    @arwes/tools@1.0.0-alpha.23 \
    @arwes/react-animated@1.0.0-alpha.23 \
    @arwes/react-animator@1.0.0-alpha.23 \
    @arwes/react-bgs@1.0.0-alpha.23 \
    @arwes/react-bleeps@1.0.0-alpha.23 \
    @arwes/react-core@1.0.0-alpha.23 \
    @arwes/react-frames@1.0.0-alpha.23 \
    @arwes/react-text@1.0.0-alpha.23 \
    @arwes/react-tools@1.0.0-alpha.23
    @arwes/react@1.0.0-alpha.23
```

Let me know if it works for you 🙌

**Comment by 1NF053C** [comment]
Do you foresee any maintainability issues w/ pinning these versions?

For example, if we want a later react-frames pkg version without losing this animation feature, could it be better for @arwes/react pkg user to try and extract this svg animation hook out of @arwes/react and into their own source? https://github.com/arwes/arwes/blob/8db5eaab6d812b4383d9cef8210128fd30f71bde/packages/react-frames/src/useFrameSVGAssemblingAnimation/useFrameSVGAssemblingAnimation.ts

It looks like the hook's interface is `MutableRefObject<SVGSVGElement | null` in and `{ onRender }` out, but I'm not familiar with SVG / Animations so I'm not too sure what could cause maintainability issues here over time.

Thoughts?

Edit: I see it depends on `import { useAnimator } from '@arwes/react-animator';`, could it be tricky to maintain compatibility with future `useAnimator` or `AnimatorContext` changes?

**Comment by romelperez** [comment]
There will be a few important breaking changes with the upcoming versions, since it's still in alpha state. Many of them are just renaming stuff.

For now if what the version you are using is enough for you, leave them pinned and when a new release is made, you can check the [release notes](https://github.com/arwes/arwes/releases) for details.

That animation functionality in particular was updated to be used like this in the upcoming version: https://next.arwes.dev/play?code=&type=predefined&sandbox=Components%7C%40arwes%2Freact-frames%7Cbasic

In other words, leave the packages pinned and wait until the next release is made to see how to migrate 🙌

**Comment by Webitect0x** [comment]
Ok great I have confirmed it works now, thank you!
